addons/point_of_sale/point_of_sale.py

```diff
@@ -1003,7 +1003,7 @@ def action_invoice(self, cr, uid, ids, context=None):
                 inv_line['price_unit'] = line.price_unit
                 inv_line['discount'] = line.discount
                 inv_line['name'] = inv_name
-                inv_line['invoice_line_tax_id'] = [(6, 0, [x.id for x in line.product_id.taxes_id] )]
+                inv_line['invoice_line_tax_id'] = [(6, 0, [x.id for x in line.tax_ids])] if line.tax_ids else [(6, 0, [x.id for x in line.product_id.taxes_id])]
```

Comment:
Could we split this line for readability? I'd use a variable or a full if/else block.